### ISSUE TITLE
Create balance check entries in the uptodate-status table/page

### ISSUE THREAD

**Issue by redstreet** [aumayr/beancount-web#70]
It would be nice to have a button that generates the beancount entries required to "promote" accounts from their current uptodate-status to the best uptodate-status (green), and copies it into the clipboard, so the user can paste it into their source and edit as required. This is helpful when the user has just updated their accounts, and sees that the balances correspond with their institution, and wants an easy way to mark all or many accounts as green (known-good).

The text copied would look like:
````
2015-12-23 balance Assets:Bank:Duck-Bank 27 USD
2015-12-23 balance Liabilities:Duck-Card -7 USD
...
````
where 27 USD and -7 USD are the current balances in the accounts. A line has to be created for each commodity in each account.

The button could be close to the table that summarizes the uptodate-statuses.

**Comment by aumayr** [comment]
@redstreet Please try if this does what it should :-)

**Comment by redstreet** [comment]
Nice, thank you! This addresses the case of generating these entries for single accounts. Feedback:

- it would be even more valuable to have a button or something that would generate and copy *all* non-green balance entries to the clipboard, in the order displayed in the table, all at once. The real value here is to "green" many accounts with minimal effort (presumably after the user  looks at all the institutions and knows the numbers match)

- the green dots seem to behave weirdly: the first time I click on the dot, nothing is copied. But if I click on a yellow, then back on a green, it copies. Should the green dots have this functionality at all? (I'm leaning towards not)

I'm curious: have you found this feature useful if you used it?

**Comment by redstreet** [comment]
One other major point: there needs to be a current balance column in the table. Else, it's not easy to tell whether or not an account needs to be upgraded to green. I can see the balance appears on hovering, but that's too painful to use if I have lots of accounts that need attention.

**Comment by aumayr** [comment]
Thanks @redstreet for testing and feedback! 

> it would be even more valuable to have a button or something that would generate and copy all non-green balance entries to the clipboard, in the order displayed in the table, all at once. The real value here is to "green" many accounts with minimal effort (presumably after the user looks at all the institutions and knows the numbers match)

I added that feature (e1547662f245d932309d25350204553e44cf6d6f). It copies all balance directives for none-green accounts, and there is a tooltip to preview the text. It is generated in the order that is the default-order (if you change the order via clicking the table header the new ordering is not taken into account). 

> the green dots seem to behave weirdly: the first time I click on the dot, nothing is copied. But if I click on a yellow, then back on a green, it copies. Should the green dots have this functionality at all? (I'm leaning towards not)

Which browser are you using?

> I'm curious: have you found this feature useful if you used it?

I haven't had the opportunity to test it with my personal workflow yet, but I think - especially with the new button - that this could become very time-saving. 

> One other major point: there needs to be a current balance column in the table. Else, it's not easy to tell whether or not an account needs to be upgraded to green. I can see the balance appears on hovering, but that's too painful to use if I have lots of accounts that need attention.

Added that in e1547662f245d932309d25350204553e44cf6d6f as well.

**Comment by redstreet** [comment]
Thank you - this works beautifully! :)

I'm using Chrome on Win7. Since the green dots don't copy any more (which is good), I don't see the issue any more. A small related issue: the "known good" hover text includes the "Click here to copy" text, which should be removed, right?

**Comment by aumayr** [comment]
> Since the green dots don't copy any more (which is good), I don't see the issue any more. A small related issue: the "known good" hover text includes the "Click here to copy" text, which should be removed, right?

No, this sounds like a bug! I did not remove the functionality for the "known good"s, but it seems that it just does not work in your browser. I need to look into that, and fix it if I can reproduce it.

**Comment by redstreet** [comment]
Okay, I can try to pinpoint the exact behavior tomorrow if that will help, and also try on a different browser, and get back to you.

**Comment by aumayr** [comment]
Awesome, thanks!

**Comment by redstreet** [comment]
Works fine on IE and on a fresh Chrome profile, so I'll chalk this one up to something with my browser plugins/cache etc.

One issue: accounts that have been closed have a blank after "Click here to copy:" and do not copy anything. The behavior is correct, but the "Click here to copy:" text should be removed.

The "Statistics" page now takes a while to load, understandably. It's not terrible, but if at all there's some low hanging fruit here, it would be great to fix.

Otherwise, good to go, thanks for this!

**Comment by aumayr** [comment]
> One issue: accounts that have been closed have a blank after "Click here to copy:" and do not copy anything. The behavior is correct, but the "Click here to copy:" text should be removed.

Fixed in 18fee41
